cmd/limactl/inspect.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/goccy/go-yaml"
+	"github.com/lima-vm/lima/pkg/store"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func newInspectCommand() *cobra.Command {
+	inspectCommand := &cobra.Command{
+		Use:   "inspect [flags] [INSTANCE]...",
+		Short: "Inspect instances of Lima.",
+		Long: `Inspect instances of Lima.
+		Includes both the current state, as well as the combined configuration used to create the instance.
+		The output can be presented in one of several formats, using the --format <format> flag.
+		
+		  --format json - output in json format
+		  --format yaml - output in yaml format
+		  --format '{{ <go template> }}' - if the format begins and ends with '{{ }}', then it is used as a go template.
+		`,
+		Args:              cobra.ArbitraryArgs,
+		RunE:              inspectAction,
+		ValidArgsFunction: inspectBashComplete,
+	}
+
+	inspectCommand.Flags().StringP("format", "f", "json", "output format, one of: json, yaml, go-template")
+
+	return inspectCommand
+}
+
+func inspectAction(cmd *cobra.Command, args []string) error {
+	format, err := cmd.Flags().GetString("format")
+	if err != nil {
+		return err
+	}
+
+	allinstances, err := store.Instances()
+	if err != nil {
+		return err
+	}
+
+	instanceNames := []string{}
+	if len(args) > 0 {
+		for _, arg := range args {
+			matches := instanceMatches(arg, allinstances)
+			if len(matches) > 0 {
+				instanceNames = append(instanceNames, matches...)
+			} else {
+				logrus.Warnf("No instance matching %v found.", arg)
+			}
+		}
+	} else {
+		instanceNames = allinstances
+	}
+
+	// get the state and config for all the requested instances
+	var instances []*store.Instance
+	for _, instanceName := range instanceNames {
+		instance, err := store.Inspect(instanceName)
+		if err != nil {
+			return fmt.Errorf("unable to load instance %s: %w", instanceName, err)
+		}
+		instances = append(instances, instance)
+	}
+
+	// output in the requested format
+	switch {
+	case format == "json":
+		b, err := json.Marshal(instances)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(cmd.OutOrStdout(), string(b))
+	case format == "yaml":
+		b, err := yaml.Marshal(instances)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(cmd.OutOrStdout(), string(b))
```

Comment:
```go
switch format {
  case "json":
    format = "{{json .}}" // use template.FuncMap to implement the "json" func
  case "yaml":
    format = "{{yaml .}}"
}
```

Reply:
Also, the logic can be moved to `pkg/textutil` so that it can be shared across multiple commands

Reply:
Not quite, it creates a loop:

```
package github.com/lima-vm/lima/cmd/lima-guestagent
        imports github.com/lima-vm/lima/pkg/textutil
        imports github.com/lima-vm/lima/pkg/store
        imports github.com/lima-vm/lima/pkg/limayaml
        imports github.com/lima-vm/lima/pkg/networks
        imports github.com/lima-vm/lima/pkg/textutil: import cycle not allowed
```

I can put it in `pkg/store`

Reply:
I do not see the advantage of doing the `switch` that way and defining the json func, vs the way it is now with a switch?